File: modules/server-play/app/controllers/Event.scala
package controllers

import javax.xml.bind.annotation.{XmlAccessType, XmlAccessorType, XmlRootElement}

import be.wegenenverkeer.jaxb.{xmlElement, xmlAttribute, StringOptionAdapter, xmlTypeAdapter}
import play.api.libs.json._

object EventFormat {
  implicit val eventFormat = Json.format[Event]
}

@XmlRootElement
@XmlAccessorType (XmlAccessType.NONE)
case class Event(@xmlElement value: Double,
                 @xmlElement @xmlTypeAdapter(classOf[StringOptionAdapter]) description: Option[String],
                 @xmlAttribute version: Int = 1) {
    def this() = this(0.0, None, 0)

}

File: modules/client-scala/src/test/scala/be/wegenenverkeer/atomium/client/Event.java
package be.wegenenverkeer.atomium.client;

import jakarta.xml.bind.annotation.*;

// this is the Event model, annotations for XML deserialization
// and public accessors for JSON
@XmlRootElement
@XmlAccessorType(XmlAccessType.NONE)
class Event {

    @XmlElement
    public Double value;

    @XmlElement
    public String description;

    @XmlAttribute
    public Integer version;

    public Event(){
    }


    public String toString() {
        return "Event " + version + " " + "description " + " value: " + value;
    }

}
